independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java

```diff
@@ -291,12 +295,13 @@ void notify(T event, ObserverExceptionHandler exceptionHandler, boolean async) {
                             }
 
                             Synchronization sync = new ArcSynchronization(deferredEvents);
-                            TransactionManager txManager = transactionManagerInstance.get();
+                            TransactionSynchronizationRegistry tsr = transactionSynchronizationRegistryInstance.get();
+
                             try {
                                 // NOTE - We are using standard synchronization on purpose as that seems more
                                 // fitting than interposed sync. Either way will have some use-cases that won't work.
                                 // See for instance discussions on https://github.com/eclipse-ee4j/cdi/issues/467
-                                txManager.getTransaction().registerSynchronization(sync);
+                                tsr.registerInterposedSynchronization(sync);
```

Comment:
I think the comment just above is saying we shouldn't use interposed synchronization here... Can you confirm it's wrong? If so, it should probably be updated.

Reply:
> @manovotn Do you recall why you changed this to a session sync (earlier it had been interposed).

No I don't, there is a comment with a link directly above this code. You'd probably need to trace it back to see why we did it. But I am pretty sure there must have been some use case for it, I don't see us just changing it on a whim.
I can try looking around tomorrow but it's quite a while ago :)

Reply:
I thought it might be because using an interposed synchronisation meant that you couldn't guarantee when it would run and you wanted it to run before other synchronisations and using a session would enable that. But now that we can order it to run first, using an interposed synchronisation is now feasible.

Reply:
This issue is also related to interposed sync and ordering of it - https://github.com/quarkusio/quarkus/issues/18737
Namely the following comment seems relevant - https://github.com/quarkusio/quarkus/issues/18737#issuecomment-911712330
The commit performing the change as such doesn't have any valuable info though.

Reply:
I think one of the reasons for standard sync. was that the code in the observers is an arbitrary user code. So in a way, using tx observer is similar to writing and registering your own custom synchronization. And I was led to believe that interposed sync. was meant for container-based events rather than user code but my knowledge of JTA is shallow so I might be wrong :)

Having Arc use interposed sync. is probably OK except a case where user would also register their own synchronization in which case the ordering will be somewhat awkward.

> I thought it might be because using an interposed synchronisation meant that you couldn't guarantee when it would run

If it wasn't possible to order them back when the PR was done, that might well have been another reason for it.

Another thing probably worth noting is that Weld (or rather [WFLY integration of it](https://github.com/wildfly/wildfly/blob/main/weld/transactions/src/main/java/org/jboss/as/weld/services/bootstrap/WeldTransactionServices.java#L95)) always used standard synchronization too.

@mkouba any chance you recall why we went from interposed to standard in Arc?

Reply:
> @mkouba any chance you recall why we went from interposed to standard in Arc?

I'm sorry but I have no idea :-(

Reply:
> I'm sorry but I have no idea :-(

No problem.

Anyway, if a JTA expert says it should be interposed, I am not against it so long as CI ends up green :)
Just please update the comment above the code @mmusgrov

Reply:
Thanks, I think your arguments are persuasive and I've backed out the change.

I've improved the test to cause hibernate and agroal to also register synchronizations but I hit a problem mixing `@RegisterExtension` with `@QuarkusTest` in the same test suite. I reported the problme on zulip so I will delay pushing another commit until that's resolved.

Reply:
>... I've backed out the change.

Not sure I follow? I am still seeing the interposed sync in this PR

Reply:
I don't see a change to `EventImpl` anymore.